src/AudioAnalysisTools/Harmonics/HarmonicParameters.cs

```diff
@@ -33,17 +33,26 @@ public class HarmonicParameters : CommonParameters
         public int? MinFormantGap { get; set; }
 
         /// <summary>
-        /// Gets or sets the the top bound of gap between formants. Units are Hertz.
+        /// Gets or sets the top bound of gap between formants. Units are Hertz.
         /// </summary>
         public int? MaxFormantGap { get; set; }
 
+        /// <summary>
+        /// Gets or sets a smoothing window.
+        /// This is used to run a moving average filter along each of the frequency bins.
+        /// It can help to smooth over discontinuous formants.
+        /// </summary>
+        public int? SmoothingWindow { get; set; } = 0;
```

Comment:
should probably default to `null` if it is nullable

Also add validation to new properties so we can give users nice feedback for bad input

Reply:
I added new functionality due to your comment about adjusting start/end frames to compensate for smoothing. Making the smoothing explicit means I could do away with hidden adjusting. I will remove the nullable because it is best to have a default value = 0.

Reply:
How do I add validation?

Reply:
Like this:

https://github.com/QutEcoacoustics/audio-analysis/blob/9a2b560a43aa2bada23e959b3888d8a367b56a68/src/AudioAnalysisTools/Events/BlobParameters.cs#L30-L39